implementations/rust/ockam/ockam_core/Cargo.toml

```diff
@@ -32,7 +32,7 @@ no_std = ["core2", "hex", "rand_pcg", "spin"]
 alloc = ["core2/alloc", "heapless", "hex/alloc", "serde/alloc", "serde_bare/alloc"]
 
 [dependencies]
-ockam_message_derive = { path = "../ockam_message_derive", version = "0.1.0"     , default_features = false }
+ockam_macros = { path = "../ockam_macros", version = "0.1.0"     , default_features = false }
```

Comment:
Isn't there a circular dependency?

Reply:
I created the PR to run the github workflows and see if everything compiles. I tried compiling a few crates on my computer, but I haven't had the time yet to run a full test. I also want to add some context to the issue description. I'll try to do it as soon as possible. Thanks sanjo for your time!

Reply:
`ockam_macros` depend on `ockam` on `dev_dependencies`, which, as far as I know, is not a problem as the tests are run against the compiled version without the dev deps. I'm probably missing something.

Reply:
Hm, I think this can still cause issues during publish. I'd prefer you move the tests into the ockam crate, breaking the cycle.

Reply:
OK done! Now none of the macros crates has a dependency on ockam/core/node.